pandas/tests/frame/indexing/test_mask.py

```diff
@@ -99,3 +102,24 @@ def test_mask_try_cast_deprecated(frame_or_series):
     with tm.assert_produces_warning(FutureWarning):
         # try_cast keyword deprecated
         obj.mask(mask, -1, try_cast=True)
+
+
+def test_mask_stringdtype():
```

Comment:
can you make sure where have an equivlanet test for .where?

Reply:
Added.